pkg/R/types.R

```diff
@@ -0,0 +1,47 @@
+# Utility functions for handling Spark DataTypes.
+
+# Handler for StructType
+structType <- function(st) {
+  obj <- structure(new.env(parent = emptyenv()), class = "structType")
+  obj$jobj <- st
+  obj$fields <- lapply(SparkR:::callJMethod(st, "fields"), structField)
```

Comment:
Ohhhh, that's what I wanted to do but I couldn't figure out a way to do it! That'd be perfect.